tests/runtime/servicers/test_model_train_servicer_impl.py

```diff
@@ -58,11 +58,11 @@ def clear_messages_from_servicer(servicer):
 
 
 @pytest.fixture(autouse=True, params=[True, False])
-def set_train_location(request):
+def set_train_location(request, reset_model_manager):
```

Comment:
Even though we had this section here to parameterize `use_subprocess`, it doesn’t actually work since the `get_trainer` function always fetches a trainer that was initialized before this config comes into play. Hence we’re never actually checking a training in a `sub_process`. I verified this by running a failing test and seeing that we always hit `destroyable_thread.py` in the stacktrace instead of `destroyable_process` in both scenarios.

To fix this, we need to add `reset_model_manager` to make sure all trainers use the config that we provide.

Reply:
@prashantgupta24 can you add that context directly to the docstring for this fixture?